src/db/setup.ts

```diff
@@ -0,0 +1,43 @@
+import sqlite3 from 'sqlite3';
+import { open } from 'sqlite';
+import path from 'path';
+import fs from 'fs';
+
+async function setupDatabase() {
+    // Ensure the data directory exists
+    const dataDir = path.join(process.cwd(), 'data');
+    if (!fs.existsSync(dataDir)) {
+        fs.mkdirSync(dataDir, { recursive: true });
+    }
+
+    // Open database connection
+    const db = await open({
+        filename: path.join(dataDir, 'db.sqlite'),
+        driver: sqlite3.Database,
+    });
+
+    // Create users table
+    await db.exec(`
+    CREATE TABLE IF NOT EXISTS users (
+      id TEXT PRIMARY KEY,
+      firstName TEXT NOT NULL,
+      lastName TEXT NOT NULL,
+      email TEXT UNIQUE NOT NULL,
+      username TEXT UNIQUE NOT NULL,
+      registrationDate TEXT NOT NULL
+    )
+  `);
+
+    console.log('Database setup complete');
+    return db;
```

Comment:
Hopefully this works. By returning the db object we can continue to use it in other scripts. Haven't tested because of the error in the original comment